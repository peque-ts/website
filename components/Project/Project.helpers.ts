const SNIPPET_FRAMEWORK = `import { Controller, Get, Module, PequeFactory } from '@pequehq/framework';

@Controller('/test')
class MyController {
  @Get('/hello')
  hello() {
    return { test: 'hello world' };
  }
}

@Module({ controllers: [MyController] })
class MyModule {}

const webserver = PequeFactory.createServer({
  cors: true,
  isCpuClustered: false,
  rootModule: MyModule,
});

webserver.start();`;

export { SNIPPET_FRAMEWORK };
