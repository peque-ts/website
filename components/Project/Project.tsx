import clsx from 'clsx';
import Link from 'next/link';
import React from 'react';

import { Button } from '../Button';
import { Code } from '../Code';
import { Icon } from '../Icon';
import { List } from '../List';
import { Terminal } from '../Terminal';

interface Props {
  name: string;
  description: string;
  features: string[];
  docsLink: string;
  command: string;
  exampleCodePosition?: 'left' | 'right';
  exampleCode: string;
  bgClassName?: string;
  className?: string;
}

export const Project: React.VFC<Props> = ({
  name,
  description,
  features,
  docsLink,
  command,
  exampleCodePosition = 'right',
  exampleCode,
  bgClassName,
  className,
}) => {
  const renderContent = () => (
    <>
      <h1>{name}</h1>
      <h2 className="text-lg font-normal mb-6">{description}</h2>
      <List className="space-y-2" items={features}>
        {(feature) => (
          <div className="flex items-center space-x-2">
            <Icon alt={feature} name="check" size={20} />
            <span>{feature}</span>
          </div>
        )}
      </List>
      <Terminal className="my-8">{command}</Terminal>
      <Button>Check the docs</Button>
    </>
  );

  return (
    <section className={clsx('relative py-8 my-8', className)}>
      <div className={clsx('-z-10 absolute inset-0', bgClassName)} />
      <div className="container mx-auto">
        <div className="grid grid-cols-2 gap-8">
          {exampleCodePosition === 'right' ? (
            <>
              <div>{renderContent()}</div>
              <Code>{exampleCode}</Code>
            </>
          ) : (
            <>
              <Code>{exampleCode}</Code>
              <div>{renderContent()}</div>
            </>
          )}
        </div>
      </div>
    </section>
  );
};
