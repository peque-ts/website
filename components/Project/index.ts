export { Project } from './Project';
export * from './Project.helpers';
