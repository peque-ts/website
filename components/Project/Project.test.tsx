import { render } from '@testing-library/react';

import { Project } from './Project';

describe('Project', () => {
  it('should render', () => {
    const { asFragment } = render(<Project />);

    expect(asFragment()).toMatchSnapshot();
  });
});
