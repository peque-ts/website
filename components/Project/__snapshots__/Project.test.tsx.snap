// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render with example to the left 1`] = `
<DocumentFragment>
  <section
    class="relative py-8 my-8"
  >
    <div
      class="-z-10 absolute inset-0 bg-primary-800 skew-y-2"
    />
    <div
      class="container mx-auto"
    >
      <div
        class="grid grid-cols-2 gap-12"
      >
        <pre
          class="custom-scrollbar"
        >
          <code
            class="drop-shadow-xl rounded hljs language-typescript"
            role="code"
          >
            <span
              class="hljs-keyword"
            >
              import
            </span>
             { 
            <span
              class="hljs-title class_"
            >
              Controller
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              Get
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              Module
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              PequeFactory
            </span>
             } 
            <span
              class="hljs-keyword"
            >
              from
            </span>
             
            <span
              class="hljs-string"
            >
              '@pequehq/framework'
            </span>
            ;


            <span
              class="hljs-meta"
            >
              @Controller
            </span>
            (
            <span
              class="hljs-string"
            >
              '/test'
            </span>
            )

            <span
              class="hljs-keyword"
            >
              class
            </span>
             
            <span
              class="hljs-title class_"
            >
              MyController
            </span>
             {
  
            <span
              class="hljs-meta"
            >
              @Get
            </span>
            (
            <span
              class="hljs-string"
            >
              '/hello'
            </span>
            )
  
            <span
              class="hljs-title function_"
            >
              hello
            </span>
            (
            <span
              class="hljs-params"
            />
            ) {
    
            <span
              class="hljs-keyword"
            >
              return
            </span>
             { 
            <span
              class="hljs-attr"
            >
              test
            </span>
            : 
            <span
              class="hljs-string"
            >
              'hello world'
            </span>
             };
  }
}


            <span
              class="hljs-meta"
            >
              @Module
            </span>
            ({ 
            <span
              class="hljs-attr"
            >
              controllers
            </span>
            : [
            <span
              class="hljs-title class_"
            >
              MyController
            </span>
            ] })

            <span
              class="hljs-keyword"
            >
              class
            </span>
             
            <span
              class="hljs-title class_"
            >
              MyModule
            </span>
             {}


            <span
              class="hljs-keyword"
            >
              const
            </span>
             webserver = 
            <span
              class="hljs-title class_"
            >
              PequeFactory
            </span>
            .
            <span
              class="hljs-title function_"
            >
              createServer
            </span>
            ({
  
            <span
              class="hljs-attr"
            >
              cors
            </span>
            : 
            <span
              class="hljs-literal"
            >
              true
            </span>
            ,
  
            <span
              class="hljs-attr"
            >
              isCpuClustered
            </span>
            : 
            <span
              class="hljs-literal"
            >
              false
            </span>
            ,
  
            <span
              class="hljs-attr"
            >
              rootModule
            </span>
            : 
            <span
              class="hljs-title class_"
            >
              MyModule
            </span>
            ,
});

webserver.
            <span
              class="hljs-title function_"
            >
              start
            </span>
            ();
          </code>
        </pre>
        <div
          class="flex flex-col justify-center"
        >
          <h1>
            Framework
          </h1>
          <h2
            class="text-lg font-normal mb-6 text-secondary-100"
          >
            Node.js framework for backend applications.
          </h2>
          <ul
            class="space-y-1.5 mb-8"
          >
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Modular"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Modular
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Setup REST endpoints in no time"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Setup REST endpoints in no time
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Microservices"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Microservices
                </span>
              </div>
            </li>
          </ul>
          <div
            class="mb-10"
          >
            <div
              class="bg-black p-4 shadow-lg rounded inline-flex space-x-4 shadow-secondary-700"
            >
              <pre>
                <code>
                  $ npx peque new project-name
                </code>
              </pre>
              <div
                class="w-12 flex items-center justify-end"
              >
                <button
                  class="flex items-center"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Copy to clipboard
                  </span>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Copy"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div>
            <a
              class="inline-block text-sm font-semibold rounded py-3 px-8 transition bg-primary-500 shadow-lg shadow-[#E712B433] hover:shadow-[#E712B455] hover:bg-primary-400"
              href="/docs/framework"
            >
              Check the docs
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Project should render with example to the right 1`] = `
<DocumentFragment>
  <section
    class="relative py-8 my-8"
  >
    <div
      class="-z-10 absolute inset-0 bg-primary-800 skew-y-2"
    />
    <div
      class="container mx-auto"
    >
      <div
        class="grid grid-cols-2 gap-12"
      >
        <div
          class="flex flex-col justify-center"
        >
          <h1>
            Framework
          </h1>
          <h2
            class="text-lg font-normal mb-6 text-secondary-100"
          >
            Node.js framework for backend applications.
          </h2>
          <ul
            class="space-y-1.5 mb-8"
          >
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Modular"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Modular
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Setup REST endpoints in no time"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Setup REST endpoints in no time
                </span>
              </div>
            </li>
            <li>
              <div
                class="flex items-center space-x-2"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Microservices"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="text-secondary-100"
                >
                  Microservices
                </span>
              </div>
            </li>
          </ul>
          <div
            class="mb-10"
          >
            <div
              class="bg-black p-4 shadow-lg rounded inline-flex space-x-4 shadow-secondary-700"
            >
              <pre>
                <code>
                  $ npx peque new project-name
                </code>
              </pre>
              <div
                class="w-12 flex items-center justify-end"
              >
                <button
                  class="flex items-center"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Copy to clipboard
                  </span>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Copy"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div>
            <a
              class="inline-block text-sm font-semibold rounded py-3 px-8 transition bg-primary-500 shadow-lg shadow-[#E712B433] hover:shadow-[#E712B455] hover:bg-primary-400"
              href="/docs/framework"
            >
              Check the docs
            </a>
          </div>
        </div>
        <pre
          class="custom-scrollbar"
        >
          <code
            class="drop-shadow-xl rounded hljs language-typescript"
            role="code"
          >
            <span
              class="hljs-keyword"
            >
              import
            </span>
             { 
            <span
              class="hljs-title class_"
            >
              Controller
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              Get
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              Module
            </span>
            , 
            <span
              class="hljs-title class_"
            >
              PequeFactory
            </span>
             } 
            <span
              class="hljs-keyword"
            >
              from
            </span>
             
            <span
              class="hljs-string"
            >
              '@pequehq/framework'
            </span>
            ;


            <span
              class="hljs-meta"
            >
              @Controller
            </span>
            (
            <span
              class="hljs-string"
            >
              '/test'
            </span>
            )

            <span
              class="hljs-keyword"
            >
              class
            </span>
             
            <span
              class="hljs-title class_"
            >
              MyController
            </span>
             {
  
            <span
              class="hljs-meta"
            >
              @Get
            </span>
            (
            <span
              class="hljs-string"
            >
              '/hello'
            </span>
            )
  
            <span
              class="hljs-title function_"
            >
              hello
            </span>
            (
            <span
              class="hljs-params"
            />
            ) {
    
            <span
              class="hljs-keyword"
            >
              return
            </span>
             { 
            <span
              class="hljs-attr"
            >
              test
            </span>
            : 
            <span
              class="hljs-string"
            >
              'hello world'
            </span>
             };
  }
}


            <span
              class="hljs-meta"
            >
              @Module
            </span>
            ({ 
            <span
              class="hljs-attr"
            >
              controllers
            </span>
            : [
            <span
              class="hljs-title class_"
            >
              MyController
            </span>
            ] })

            <span
              class="hljs-keyword"
            >
              class
            </span>
             
            <span
              class="hljs-title class_"
            >
              MyModule
            </span>
             {}


            <span
              class="hljs-keyword"
            >
              const
            </span>
             webserver = 
            <span
              class="hljs-title class_"
            >
              PequeFactory
            </span>
            .
            <span
              class="hljs-title function_"
            >
              createServer
            </span>
            ({
  
            <span
              class="hljs-attr"
            >
              cors
            </span>
            : 
            <span
              class="hljs-literal"
            >
              true
            </span>
            ,
  
            <span
              class="hljs-attr"
            >
              isCpuClustered
            </span>
            : 
            <span
              class="hljs-literal"
            >
              false
            </span>
            ,
  
            <span
              class="hljs-attr"
            >
              rootModule
            </span>
            : 
            <span
              class="hljs-title class_"
            >
              MyModule
            </span>
            ,
});

webserver.
            <span
              class="hljs-title function_"
            >
              start
            </span>
            ();
          </code>
        </pre>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
